rfc/rfc017-distributed-network-grpc-v2.md

```diff
@@ -199,9 +220,26 @@ In the protocol described here, this would mean that an entire range of transact
 When this happens randomly, it wouldn't be such a problem, but it's also an attack vector.
 An attacker could craft pairs of transactions that would trigger other nodes into requesting large amounts of transactions. 
 This type of attack is called the [birthday attack](https://en.wikipedia.org/wiki/Birthday_attack).
-The only way to protect against this type of attack is to make the set large enough that the cost to the attacker would become to big.
-For an IBLT with 64 buckets and 4 hashes, the change for a collision is 50% after adding ~5000 keys.
+The only way to protect against this type of attack is to make the set large enough that the cost to the attacker would become to high.
+For an IBLT with 64 buckets and 4 hashes, the chance for a collision is 50% after adding ~5000 keys.
 This would be trivial for current computers.
 Expanding this to 1024 buckets and 6 hashes, reaching a 50% chance for a collision is only achieved after adding more than a billion keys.
 This is still not much, but the keys used for the collision have to be created by real signed transactions.
 Even on modern hardware, this could take an hour.
+
+### A.2 The right message for the right situation
+
+The protocol uses a variety of message to synchronize all transactions over nodes.
+Together they cover all known situations.
+
+If all nodes are operating correctly, the gossip protocol is the most efficient in synchronizing all the transactions.
+It will not be able to deal with situations where a node is offline or when nodes can no longer reach each other.
+It can be used frequent and uses small messages.
+
+If a node has been offline for some time, the combination of an IBLT and a range query will synchronize the node efficiently.
+The IBLT will resolve all missing transactions for the node's latest page and the range query will sycnhronize all the missing pages. Those pages will include all transactions that have been created during the period the nodes was offline.
+
+If a node has not been offline, but its communication has been interrupted, it will have created new transactions.
```

Comment:
_might_ have created new transactions that aren't synchronized with the rest of the network